web/modules/custom/custom_page/src/CustomPageOgMenuLinksUpdater.php

```diff
@@ -0,0 +1,125 @@
+
+  /**
+   * The menu link manager service.
+   *
+   * @var \Drupal\Core\Menu\MenuLinkManagerInterface
+   */
+  protected $menuLinkManager;
+
+  /**
+   * The menu link content storage.
+   *
+   * @var \Drupal\Core\Entity\Sql\SqlContentEntityStorage
+   */
+  protected $menuLinkContentStorage;
+
+  /**
+   * Builds a new custom page OG menu links updater service.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Menu\MenuLinkManagerInterface $menu_link_manager
+   *   The menu link manager service.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, MenuLinkManagerInterface $menu_link_manager) {
+    $this->ogMenuInstanceStorage = $entity_type_manager->getStorage('ogmenu_instance');
+    $this->menuLinkContentStorage = $entity_type_manager->getStorage('menu_link_content');
+    $this->menuLinkManager = $menu_link_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addLink(NodeInterface $custom_page) {
+    if ($og_menu_instance = $this->getOgMenuInstance($custom_page)) {
+      MenuLinkContent::create([
+        'title' => $custom_page->label(),
+        'menu_name' => 'ogmenu-' . $og_menu_instance->id(),
+        'link' => ['uri' => 'internal:/node/' . $custom_page->id()],
+        // The 'exclude_from_menu' property is used as a hidden API trick to
+        // allow a disabled menu item.
+        'enabled' => empty($custom_page->exclude_from_menu),
+      ])->save();
+    }
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function deleteLinks(NodeInterface $custom_page) {
+    /** @var \Drupal\Core\Menu\MenuLinkTreeInterface $t */
+    if ($og_menu_instance = $this->getOgMenuInstance($custom_page)) {
+      $menu_name = "ogmenu-{$og_menu_instance->id()}";
+      foreach ($custom_page->uriRelationships() as $rel) {
+        $url = $custom_page->toUrl($rel);
+        // Delete all MenuLinkContent links that point to this entity route.
+        if ($result = $this->menuLinkManager->loadLinksByRoute($url->getRouteName(), $url->getRouteParameters())) {
+          /** @var \Drupal\menu_link_content\Plugin\Menu\MenuLinkContent $instance */
+          foreach ($result as $id => $instance) {
+            if ($instance->getMenuName() === $menu_name && $instance->isDeletable() && strpos($id, 'menu_link_content:') === 0) {
+              $instance->deleteLink();
+              // Search for children of deleted menu link.
+              $mids = $this->menuLinkContentStorage->getQuery()
+                ->condition('parent', "menu_link_content:{$instance->getDerivativeId()}")
+                ->execute();
+              if ($mids) {
+                // Remove the relationship to the deleted parent menu link.
+                foreach (MenuLinkContent::loadMultiple($mids) as $menu_link_content) {
+                  $menu_link_content->set('parent', NULL)->save();
+                }
```

Comment:
I'm not entirely sure if this is desired on a functional level. What is expected by the end user if a parent item is being moved to a new collection? Do the child items move along? Or should they be orphaned like is happening here?

This situation is not described in the original ticket, should we discuss this with the functional team first?